dev/com.ibm.ws.container.service/src/com/ibm/ws/container/service/state/ApplicationStateListenerConstants.java

```diff
@@ -0,0 +1,28 @@
+/*******************************************************************************
+ * Copyright (c) 2023 IBM Corporation and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License 2.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-2.0/
+ * 
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ *******************************************************************************/
+package com.ibm.ws.container.service.state;
+
+public class ApplicationStateListenerConstants {
+
+    //This class stores all the service rankings for application state listeners in one place alongside the reasons for them to have that rank.
+
+    public static final int DEFAULT_SERVICE_RANK = 0;
+
+    //EJB must shut down after CDI (see below).
+    public static final int EJB_SERVICE_RANK = DEFAULT_SERVICE_RANK;
+
+    //CDI must shut down before EJB as EJBs can have a cdi application scope and according to the CDI spec "jakarta.enterprise.event.Shutdown is not after @BeforeDestroyed(ApplicationScoped.class)"
+    public static final int CDI_SERVICE_RANK = 1;
+
+    //CDI can trigger application code during startup. This application code may do JNDI lookups, thus injection engine must start before CDI so it can provide JNDI results.
+    public static final int INJECTION_ENGINE_SERVICE_RANK = 2;
+
+}
```

Comment:
Multiple issues with this new class.
1. `com.ibm.ws.container.service.state` is exposed as an SPI.
  a. These constants are (for the moment) internal so this class should be put in a different package.
  b. If we did want to expose this class as SPI in the future then it needs proper JavaDoc.
2. We should leave some gaps in between these service ranks so that others can be added in future. For example, make CDI=100 and Injection Engine=200.